dss/storage/checkout/__init__.py

```diff
@@ -34,11 +34,18 @@ def validate_file_dst(replica: Replica, dst_bucket: str, dst_key: str):
 
 def pre_exec_validate(
         replica: Replica, dss_bucket: str, dst_bucket: str, bundle_uuid: str, bundle_version: str) -> bool:
+    validate_dst_name(dst_bucket)
     validate_dst_bucket(replica, dst_bucket)
     validate_bundle_exists(replica, dss_bucket, bundle_uuid, bundle_version)
     return True
 
 
+def validate_dst_name(dst_bucket: str)-> bool:
```

Comment:
The usage of `DestinationBucketNotFoundError` to propagate this error is misleading. The destination did not meet the expected format, so the bucket could not be discovered.

Consider performing this check in `dss/api/bundles/checkout.py` so that `DestinationBucketNotFoundError` is not needed.

Reply:
I had placed the function within this file because it has other functions that are used within testing for dst_bucket validation. I have a work around, but I think the testing approach would have to be different from the other bucket tests that are in place.

Reply:
check moved to dss/api/bundles/checkout.py